### PR TITLE
[ACA-2755] [SSO] Unable to access a private url after a public url was loaded in the same browser tab

### PR DESCRIPTION
**Pull request by mmuller88**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [x] Bugfix
> - [ ] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)
See bug description. After loading public urls like /settings you couldn't change to other urls properly in the SSO config

**What is the new behaviour?**
Now it is forwarding you to login.

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [ ] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:

### CODE REVIEW

**Comment by mmuller88** [comment]
@eromano thx for the feedback. I will try to use the existing js api method as it is just a copy of that. Regarding unit test I will see how much efforts that would be. But you know best it is SSO + silentlogin + Authguard + public / private url :-/

**Comment by mmuller88** [comment]
@eromano Requested changes 1) using js-api 2) unit test are done pls review

File: lib/core/services/auth-guard-base.ts [hunk 1]
```diff
@@ -70,6 +70,10 @@ export abstract class AuthGuardBase implements CanActivate, CanActivateChild {
             this.dialog.closeAll();
         }
 
+        if (this.isSilentLogin() && this.oauth2Auth && !this.oauth2Auth.isPublicUrl()) {
```
Comment (eromano) [inline lib/core/services/auth-guard-base.ts]: this method has only one exit point so to make it more readable can we keep it?

Comment (mmuller88) [reply lib/core/services/auth-guard-base.ts]: Sure but I doubt that it is more readable. I still assume that some of the SSO stuff shouldn't live in alfresco-js-api at all :-/

File: lib/core/services/auth-guard-base.ts [hunk 2]
```diff
@@ -61,7 +62,7 @@ export abstract class AuthGuardBase implements CanActivate, CanActivateChild {
             this.dialog.closeAll();
         }
 
-        return checkLogin;
+        return true;
```
Comment (pionnegru) [inline lib/core/services/auth-guard-base.ts]: correct me if Im wrong, but this means that any other guards extending this base service, with its own 'checkLogin' implementation, [example this one](https://github.com/Alfresco/alfresco-ng2-components/blob/development/lib/core/services/auth-guard-ecm.service.ts#L40), event if it returns `false` wi discard it and return true?

**Pull request closed without merging**